yt_dlp_plugins/postprocessor/stash.py

```diff
@@ -72,4 +79,214 @@ def run(self, info):
+                
+                if not existing_studio:
+                    existing_studio = self.stash.find_studios({"aliases": {"modifier": "EQUALS", "value": studio_name}})
+                
+                if existing_studio and len(existing_studio) > 0:
+                    update_scene["studio_id"] = existing_studio[0]["id"]
+                    self.to_screen(f"[Debug] using existing studio {existing_studio}")
+                else:
+                    # Create the studio if it doesn't exist
+                    self.to_screen("[Debug] creating new studio")
+                    new_studio = {
+                        "name": studio_name,
+                        "url": scene_data["studio"]["url"] if scene_data["studio"].get("url") else None
+                    }
+                    created_studio = self.stash.create_studio(new_studio)
+                    update_scene["studio_id"] = created_studio["id"]
+
+            self.to_screen(f"[Debug] Scene update payload: {update_scene}")
+
+            try:
+                self.stash.update_scene(update_scene)
+                self.to_screen("[Debug] Scene updated with scraped metadata.")
+            except Exception as e:
+                self.to_screen(f"[Error] Error updating scene metadata: {str(e)}")
+
+        except Exception as e:
+            self.to_screen(f"[Error] Unexpected error during processing: {str(e)}")
+
+        return [], info
+    
+    def scrape_scene_by_url(self, url):
+        query = """
+        query scrapeSceneByURL($url: String!) {
+            scrapeSceneURL(url: $url) {
+                title
+                date
+                details
+                tags {
+                    name
+                }
+                performers {
+                    name
+                    url
+                }
+                studio {
+                    name
+                    url
+                }
+            }
+        }
+        """
+        variables = {
+            "url": url
+        }
+        try:
+            self.to_screen(f"[Debug] Sending GraphQL scrapeSceneByURL query for URL: {url}")
+            response = self.stash.call_GQL(query, variables)
+            self.to_screen(f"[Debug] Full GraphQL response: {response}")
+
+            # Adjust the response check to accommodate both formats
+            scrape_scene_data = None
+            if 'data' in response and isinstance(response['data'], dict):
+                scrape_scene_data = response['data'].get('scrapeSceneURL')
+            elif 'scrapeSceneURL' in response:
+                scrape_scene_data = response['scrapeSceneURL']
+
+            if scrape_scene_data is None:
+                self.to_screen("[Error] 'scrapeSceneURL' field missing or is None in GraphQL response.")
+                self.to_screen(f"[Debug] Response structure: {response}")
+                return None

```

Comment:
```suggestion
            if scrape_scene_data is None:
                self.to_screen("[Error] 'scrapeSceneURL' field missing or is None in GraphQL response.")
                self.to_screen(f"[Debug] Response structure: {response}")
                return None
```
Shouldn't this already be covered by the if else before?